source/screens/LoginScreen.js

```diff
@@ -122,86 +191,155 @@ function LoginScreen(props) {
+          <Form>
+            <Button z={0} size="large" block onClick={() => handleLogin()}>
+              <Text>Logga in med Mobilt BankID</Text>
+            </Button>
+            <Link onPress={() => setModalVisible(true)}>Fler alternativ</Link>
+          </Form>
+        )}
+
+        <Footer>
+          <FooterText>
+            När du använder tjänsten Mitt Helsingborg godkänner du vårt{' '}
+            <ParagraphLink
+              onPress={() =>
+                Linking.openURL(
+                  'https://helsingborg.se/toppmeny/om-webbplatsen/sa-har-behandlar-vi-dina-personuppgifter/'
+                )
+              }
+            >
+              användaravtal
+            </ParagraphLink>{' '}
+            och att du har tagit del av hur vi hanterar dina{' '}
+            <ParagraphLink
+              onPress={() =>
+                Linking.openURL(
+                  'https://helsingborg.se/toppmeny/om-webbplatsen/sa-har-behandlar-vi-dina-personuppgifter/'
+                )
+              }
+            >
+              personuppgifter
+            </ParagraphLink>
+            .
+          </FooterText>
+        </Footer>
+      </Body>
+
+      <LoginModal
+        visible={modalVisible}
+        scrollViewProps={{
+          keyboardShouldPersistTaps: 'handled',
+          contentContainerStyle: { flexGrow: 1 },
+          extraScrollHeight: 50,
+        }}
+      >
+        <CloseModalButton onClose={() => setModalVisible(false)} showBackButton={false} />
+        <Body>
+          <Header>
+            <ModalHeading>Logga in med BankID på en annan enhet</ModalHeading>
+            <ContentText>
+              Öppna Mobilt BankID eller BankID på din andra enhet innan du trycker på logga in här
+              nedaför.
+            </ContentText>
+          </Header>
+
+          {(isLoading || isResolved) && (
+            <Form>
+              <AuthLoading
+                colorSchema="blue"
+                isResolved={isResolved}
+                cancelSignIn={() => handleCancelOrder()}
+                isBankidInstalled={false}
+              />
+            </Form>
+          )}
+
+          {(isIdle || isRejected) && (
+            <Form>
+              <Label strong>PERSONNUMMER</Label>
+              <LoginInput
+                placeholder="ååååmmddxxxx"
                 value={personalNumber}
                 onChangeText={handlePersonalNumber}
                 keyboardType="number-pad"
-                returnKeyType="done"
                 maxLength={12}
-                onSubmitEditing={handleSubmit}
+                onSubmitEditing={() => handleLogin(true)}
                 center
               />
-            </LoginFormField>
-
-            <LoginFormField>
-              <Button color="purpleLight" block onClick={handleSubmit}>
-                <Text>Logga in med mobilt BankID</Text>
+              <Button
+                z={0}
+                disabled={isButtonDisabled}
```

Comment:
This can be replaced with disabled={personalNumber.length !== 12} and then you don't need the isButtonDisabled or the useEffect for this.

Reply:
That is correct, not sure what i was thinking